spanner/json_column_type_samples.rb

```diff
@@ -0,0 +1,129 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "./utils"
+
+def add_json_column project_id:, instance_id:, database_id:
+  # [START spanner_add_json_column]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  db_admin_client = \
+    Google::Cloud::Spanner::Admin::Database::V1::DatabaseAdmin::Client.new
+
+  database_path = db_admin_client.database_path project: project_id,
+                                                instance: instance_id,
+                                                database: database_id
+
+  statements = ["ALTER TABLE Venues ADD COLUMN VenueDetails JSON"]
+  job = db_admin_client.update_database_ddl database: database_path,
+                                            statements: statements
+  job.wait_until_done!
+
+  puts "Added VenueDetails column to Venues table in database #{database_id}"
+  # [END spanner_add_json_column]
+end
+
+def update_data_with_json_column project_id:, instance_id:, database_id:
+  # [START spanner_update_data_with_json_column]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client  = spanner.client instance_id, database_id
+
+  rows = [{
+    VenueId: 1,
+    VenueDetails: { rating: 9, open: true }
+  }]
+  client.update "Venues", rows
+
+  # VenueDetails must be specified as a string, as it contains a top-level
+  # array of objects that should be inserted into a JSON column. If we were
+  # to specify this value as an array instead of a string, the client
+  # library would encode this value as ARRAY<JSON> instead of JSON.
+  rows = [{
+    VenueId: 2,
+    VenueDetails: [
+      {
+        name: "room 1",
+        open: true
+      },
+      {
+        name: "room 2",
+        open: false
+      }
+    ].to_json
```

Comment:
It would be more clear, then, to create the string separately. For example:

```ruby
venue_details_string = [
  {
    name: "room 1",
    open: true
  },
  {
    name: "room 2",
    open: false
  }
].to_json

rows = [
  {
    VenueId: 2,
    VenueDetails: venue_details_string
  }
]
```

Reply:
@jiren Can you please fix this issue?

Reply:
done